compiler/src/iree/compiler/Codegen/Common/ConvolutionToIGEMM.cpp

```diff
@@ -26,10 +28,13 @@ namespace {
 
 using iree_compiler::IREE::LinalgExt::IREELinalgExtDialect;
 
+/// Pattern to set a lowering configuration on an IGEMM convolution. Searches
+/// for a contraction with a linalg_ext.im2col producer, and calls the configFn
+/// to set the configuration.
 struct SetIGEMMConfiguration final : OpRewritePattern<linalg::GenericOp> {
```

Comment:
I'm not sure why this is a pattern. Shouldn't this just be a free function that returns a producer linalg_ext.im2col and you can set lowering config on it?

I guess not for this PR since it's already plumbed in, but this is weird.

I generally would use patterns for something that can recursively run, or can compose with some other patterns and recursively run. This pattern is a one off function.

Reply:
This makes sense, since it is just setting an attribute. I can refactor this pattern in another PR.